### ISSUE TITLE
Wrong imgsz on torch distributed run

### ISSUE THREAD

**Issue by akkolyth** [ultralytics/ultralytics#2893]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Running yolov8 on multiple GPU's (2 in my case) ignores fact, that imgsz=1280.  

### Environment

- Ultralytics YOLOv8.0.110 🚀 
- Python-3.8.10 torch-1.13.0+cu117 CUDA:0 (NVIDIA GeForce RTX 3090, 24260MiB) CUDA:1 (NVIDIA GeForce RTX 3090, 24260MiB)

### Minimal Reproducible Example

Running:
```
!yolo classify train \
    data=/home/jovyan/work/MDD/datasets/EXPERIMENT \
    model=/home/jovyan/work/MDD/models/weights/yolov8x-cls.pt \
    name=/home/jovyan/work/MDD/experiments/exp_$EXPERIMENT_NAME \
    plots=True \
    verbose=True \
    epochs=5 \
    patience=10 \
    batch=8 \
    device=0,1 \
    imgsz=1280
```

WARNING ⚠️ Upgrade to torch>=2.0.0 for deterministic training.
DDP settings: RANK 0, WORLD_SIZE 2, DEVICE cuda:0
TensorBoard: Start with 'tensorboard --logdir /home/jovyan/work/MDD/experiments/exp_train_PUBLIC_val__baseline12', view at http://localhost:6006/
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
optimizer: SGD(lr=0.01) with parameter groups 50 weight(decay=0.0), 51 weight(decay=0.0005), 51 bias
albumentations: RandomResizedCrop(p=1.0, height=224, width=224, scale=(0.5, 1.0), ratio=(0.75, 1.3333333333333333), interpolation=1), HorizontalFlip(p=0.5), ColorJitter(p=0.5, brightness=[0.6, 1.4], contrast=[0.6, 1.4], saturation=[0.30000000000000004, 1.7], hue=[-0.015, 0.015]), Normalize(p=1.0, mean=(0.0, 0.0, 0.0), std=(1.0, 1.0, 1.0), max_pixel_value=255.0), ToTensorV2(always_apply=True, p=1.0, transpose_mask=False)
Image sizes 224 train, 224 val

(...) As you can see, `Image sizes 224 train, 224 val` is 224 (but config imgsz=1280).

On the other hand if I run training with device=0, everything works as expected.
(...)
`Image sizes 1280 train, 1280 val`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@akkolyth yes I'm able to reproduce this issue with 

```
yolo train classify data=imagenet10 imgsz=128 device=0,1 epochs=1
```

<img width="895" alt="Screenshot 2023-05-29 at 19 08 30" src="https://github.com/ultralytics/ultralytics/assets/26833433/1b29759c-5477-45d8-b678-7069b9bc68d7">

**Comment by glenn-jocher** [comment]
@akkolyth great news 😃! Your original issue may now be resolved ✅ in PR #2897. This will be included in `ultralytics` 8.0.111 tomorrow. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
